include/oneapi/tbb/concurrent_queue.h

```diff
@@ -131,8 +133,43 @@ class concurrent_queue {
         queue_allocator_traits::destroy(my_allocator, my_queue_representation);
         r1::cache_aligned_deallocate(my_queue_representation);
     }
+  
+    concurrent_queue& operator=( const concurrent_queue& other ){
+      if (my_queue_representation != other.my_queue_representation) {
+	my_queue_representation = other.my_queue_representation;
+      }
+      return *this;
+    }
+
+    concurrent_queue& operator=( concurrent_queue&& other ){
+      if (my_queue_representation != other.my_queue_representation) {
+	my_queue_representation = std::move(other.my_queue_representation);
+      }
+      return *this;
+    }
+
+    concurrent_queue& operator=( std::initializer_list<value_type> init ){
+      assign(init);
+      return *this;
+    }
 
+    template <typename InputIterator>
+    void assign( InputIterator first, InputIterator last ){
+      concurrent_queue src(first, last);
+      my_queue_representation->assign(*src.my_queue_representation, my_allocator, move_construct_item);
+    }
+
+    void assign( std::initializer_list<value_type> init ){
+      assign(init.begin(), init.end());
```

Comment:
It seems like an issue here and in `assign(Iterator, Iterator)`. I have checked `my_queue_representation->assign` and it seems like it does not remove all of the items from the container and immediately starts to assign. I did not deeply investigated `micro_queue::assign` method. Could you please check that such an assignment would not result in memory leak?